orchagent/portsorch.cpp

```diff
@@ -1813,7 +1820,48 @@ void PortsOrch::doPortTask(Consumer &consumer)
                         SWSS_LOG_ERROR("Unknown fec mode %s", fec_mode.c_str());
                     }
                 }
+ 
+                if (learn_mode != "")
+                {
+                    if (learn_mode == "enabled" || learn_mode == "disabled")
+                    {
+                        int32_t mode;
 
+                        if (learn_mode == "enabled")
+                        {
+                            mode = SAI_BRIDGE_PORT_FDB_LEARNING_MODE_HW;
+                        }
+                        else
+                        {
+                            mode = SAI_BRIDGE_PORT_FDB_LEARNING_MODE_DISABLE;
+                        }
+
+                        if (p.m_bridge_port_id != SAI_NULL_OBJECT_ID)
+                        {
+                            if(setBridgePortLearnMode(p, mode))
+                            {
+                                SWSS_LOG_NOTICE("Set port %s learn mode to %s", alias.c_str(), learn_mode.c_str());
+                            }
+                            else
+                            {
+                                SWSS_LOG_ERROR("Failed to set port %s learn mode to %s", alias.c_str(), learn_mode.c_str());
+                                it++;
+                                continue;
+                            }
+                        }
+                        else
+                        {
+                            m_portSetLearnModeList[alias] = learn_mode;
```

Comment:
could you add this to the current `port` class and later check if the variable `learn_mode` has value or not?

Reply:
yes, we will move learn_mode to port class. @stcheng